src/tools/illink/src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -102,7 +102,7 @@ protected override bool ReportSpecialIncompatibleMembersDiagnostic (OperationAna
 		}
 
 		protected override bool VerifyAttributeArguments (AttributeData attribute) => attribute.ConstructorArguments.Length == 0 ||
-			attribute.ConstructorArguments.Length >= 1 && attribute.ConstructorArguments[0] is { Type.SpecialType: SpecialType.System_String } ctorArg;
+			attribute.ConstructorArguments.Length >= 1 && attribute.ConstructorArguments[0] is { Type.SpecialType: SpecialType.System_String };
```

Comment:
Do we want to use a collection/range expression here (could be used in all cases)?

```suggestion
			attribute.ConstructorArguments is [ { Type.SpecialType: SpecialType.System_String }, ...];
```

Reply:
Right, this is in an analyzer/fixer. If you add a `<Compile>` item to include the `Index.cs` that's in CoreLib, that should enable using these types of expressions (I recently made that file also work as a polyfill).

Reply:
We can also delay this work to the future. It's not required to get this PR in.